src/topologies/halfedge.jl

```diff
@@ -332,45 +331,84 @@ Base.convert(::Type{HalfEdgeTopology}, t::Topology) = HalfEdgeTopology(collect(e
 # HELPER FUNCTIONS
 # -----------------
 
-function adjsort(elems::AbstractVector{<:Connectivity})
-  # initialize list of adjacent elements
-  # with first element from original list
-  list = indices.(elems)
-  adjs = Tuple[popfirst!(list)]
-
-  # the loop will terminate if the mesh
-  # is manifold, and that is always true
-  # with half-edge topology
-  while !isempty(list)
-    # lookup all elements that share at least
-    # one vertex with the last adjacent element
-    found = false
-    vinds = last(adjs)
-    for i in vinds
-      einds = findall(e -> i ∈ e, list)
-      if !isempty(einds)
-        # lookup all elements that share at
-        # least two vertices (i.e. edge)
-        for j in sort(einds, rev=true)
-          if length(vinds ∩ list[j]) > 1
-            found = true
-            push!(adjs, popat!(list, j))
-          end
-        end
+# permutation of elements in adjacent-first order
+function adjsortperm(elems::AbstractVector{<:Connectivity})
+  reduce(vcat, conneccomps(elems))
+end
+
+# connected components from list of elements
+function conneccomps(elems::AbstractVector{<:Connectivity})
+  # initialize list of connected components
+  comps = [Int[firstindex(elems)]]
+
+  # initialize list of seen vertices
+  seen = Set(indices(first(elems)))
```

Comment:
I tried to simplify the code in a few places without looking into Cthulhu.jl outputs. Please feel free to comment if you see opportunities to improve performance further.

Reply:
The previous manual `seen` initialization is faster vs `seen = Set(indices(first(elems)))`:
```
BenchmarkTools.TrialJudgement: 
  time:   -17.02% => improvement (5.00% tolerance)
  memory: -0.21% => invariant (1.00% tolerance)
```